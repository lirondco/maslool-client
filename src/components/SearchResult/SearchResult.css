.search_result {
    margin: 2%;
}

.search_result a {
    text-decoration: underline;
}

.search_result a h4:hover {
    color: red !important;
}

.search_result a:hover {
    color: red !important;
}

.search_result a h4:active {
    color: red !important;
}

.search_result a:active {
    color: red !important;
}