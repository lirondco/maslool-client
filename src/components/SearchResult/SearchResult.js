import React, { Component } from 'react'
import { Link } from 'react-router-dom'
import './SearchResult.css'
import StarRatings from 'react-star-ratings'

export default class SearchResult extends Component {
    render() {
        return (
            <div className='search_result'>
                <Link to={`/trails/${this.props.id}`}>
                    <p>{this.props.name}</p>
                </Link>
                <StarRatings 
                    rating={this.props.rating}
                    starDimension='15px'
                    starSpacing='5px'
                    starRatedColor='yellow'
                    starEmptyColor='grey'
                />
                <p>Location: {this.props.address_line} {this.props.city}, {this.props.region} {this.props.postal_code}</p>
                <p>Difficulty: {this.props.difficulty}</p>
            </div>
        )
    }
}   