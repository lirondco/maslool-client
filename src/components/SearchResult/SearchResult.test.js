import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import SearchResult from './SearchResult'

describe(`SearchResult component`, () => {
    const props ={
        name: 'test name',
        webiste: 'www.test.com',
        rating: 4,
        difficulty: 'Beginner',
        address_line: '111 Test St',
        city: 'Test Town',
        region: 'Massachutests',
        postal_code: '11111'
    }

    it('renders a search result', () => {
        const wrapper = shallow(<SearchResult {...props} />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})