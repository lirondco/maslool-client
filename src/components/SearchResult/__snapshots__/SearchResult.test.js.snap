// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResult component renders a search result 1`] = `
<div
  className="search_result"
>
  <Link
    to="/trails/undefined"
  >
    <h4>
      test name
    </h4>
  </Link>
  <StarRatings
    changeRating={null}
    gradientPathName=""
    ignoreInlineStyles={false}
    numberOfStars={5}
    rating={4}
    starDimension="15px"
    starEmptyColor="grey"
    starHoverColor="rgb(230, 67, 47)"
    starRatedColor="red"
    starSpacing="5px"
    svgIconPath="m25,1 6,17h18l-14,11 5,17-15-10-15,10 5-17-14-11h18z"
    svgIconViewBox="0 0 51 48"
    typeOfWidget="Star"
  />
  <p>
    Location: 
    111 Test St
     
    Test Town
    ,
     
    Massachutests
     
    11111
  </p>
  <p>
    Difficulty: 
    Beginner
  </p>
</div>
`;
